turnpike/plugin.py

```diff
@@ -30,13 +35,48 @@ class PolicyContext:
     * `data` is an arbitrary dictionary. If two plugins need to share data in a
       way that does not get returned to the client, putting that data in this
       dictionary is the proper way to go.
+
+    * `log_data` is a dictionary containing Elasic Common Schema compatible fields.
+      It should not be necessary to write to this structure.
+
+    * `result` is a string containing the result of the policy request for user
+      consumption. This may be set by plugins to contain information about a failure.
+
+    * `client_ip` is the calculated client IP address, taking into account CDN
+      configuration. It is an ipaddress.IPV4Address or ipaddress.IPV6Address as
+      appropriate.
     """
 
     backend = None
     auth = None
     headers = {}
     status_code = None
     data = {}
+    log_data = {}
+    result = ""
+    client_ip = None
+
+    def __init__(self):
```

Comment:
Just curious the motivation for moving this here?

Reply:
Which "this" are you referring to?

Reply:
The block of "hops" logic.

Reply:
Oh. It was previously in `source_ip` but we want to log the source IP, regardless of whether the `source_ip` plugin is used on a particular route (or even installed in the pipeline). So instead of trying to calculate it twice, I moved it to the Context and do the calculation there, making the result available for `source_ip` to access.